sources/osgShadow/ShadowMap.js

```diff
@@ -236,7 +236,7 @@ define( [
 
         // Overridable Visitor so that user can override the visitor to enable disable
         // in its own shadowmap implementation
-        this._removeNodesNeverCastingVisitor = new RemoveNodesNeverCastingVisitor( this._castsShadowTraversalMask );
+        this._removeNodesNeverCastingVisitor = settings.userCallbackNodeCaster || new RemoveNodesNeverCastingVisitor( this._castsShadowTraversalMask );
```

Comment:
I started with only the apply of the default visitor calling a callback and in the end wasn't flexible enough, so I put the entire thing as a visitor.

But it could be a callback and give the shadow casted scene root node as parameter, and then user decide what he does with it.